### ISSUE TITLE
Create a guide on working with kafka

### ISSUE THREAD

**Issue by fachat** [OpenLiberty/guides-common#293]
Create a guide how to integrate kafka with OpenLiberty.
Potential use cases:

- A "universal" web service that can be called via REST or via kafka
- A web service that sends change events (Sending a Command in the meaning of CQRS) to a kafka queue
- A kafka listener that receives kafka events and updates a database model from it (receiving a CQRS command and updating the Query model of CQRS)

Note: there is an issue about reactive kafka, but I think the issue here is with a more clearly defined scope

**Comment by yeekangc** [comment]
Thank you, @fachat.  We will see how we can reconcile the two.

**Comment by yeekangc** [comment]
@fachat, can you elaborate on what do you mean by a "universal" web service?  Thanks.

**Comment by fachat** [comment]
Absolutely. I would like to write a web service that is callable by REST/HTTP just as well as via an event/message coming in through a queue. "universal", as it does not depend on the transport method.

I probably far-fetched kind of vision would be something like 
`
@Path("/foo/bar")     // REST/HTTP parameter setting
@Queue(inputQueue="inq", outputQueue="outq")    // kafka parameter setting
class {
     ...
    @POST
    public RetVal someRestMethod(InputType inputObject) {
        ...
    }
}
`
That would be callable via REST/HTTP, as well as via a message on some input queue "inq", sending the result to some output queue "outq" (where there probably is some further message broker definition configured elsewhere).

Of course this kind of gets more complicated when there are multiple methods with different paths in the class, as you need some kind of selector on e.g. the input object coming from a kafka queue to see which method to call, but that can be done (e.g. configuring a "Selector" function( JsonObject -> String ) somewhere). That could be done in a second step maybe. 

Ok, that probably requires some more infrastructure or even a separate Microprofile spec, for the guide I would be content with a how to configure / build a class that receives a message from a queue, unmarshalls it, calls the web service from above, and sends the result into some other queue.

Maybe really far fetched, I would be content for now if the other two points related to CQRS are covered for now.

**Comment by yeekangc** [comment]
Thank you for the elaboration, @fachat.  We are looking at this.  It may turn out to be a series of guides.

Obviously, any and all contribution is always welcome and appreciated.  =)

FYI, @gkwan-ibm @proubatsis.

**Comment by hutchig** [comment]
This would obviously use the built in Liberty annotations for this @Incoming and @Outgoing I imagine: see https://openliberty.io/blog/2019/09/13/microprofile-reactive-messaging-19009.html#mpreactive and the "liberty-kafka" connector.
